modules/components/AsyncConnect.js

```diff
@@ -56,7 +56,25 @@ export default class AsyncConnect extends Component {
   }
 
   isLoaded() {
-    return this.context.store.getState().reduxAsyncConnect.loaded;
+    const componentKeys = this.props.components.reduce((componentKeysMemo, component) => {
+      if (component && component.reduxAsyncConnect && component.reduxAsyncConnect.length) {
+        const asyncConnectComponentKeys = component.reduxAsyncConnect
+          .filter(asyncConnectComponent => !!asyncConnectComponent.key)
+          .map(asyncConnectComponent => asyncConnectComponent.key);
+
+        return [...componentKeysMemo, ...asyncConnectComponentKeys];
+      }
+
+      return componentKeysMemo;
+    }, []);
+
+    const loadState = this.context.store.getState().reduxAsyncConnect.loadState;
+    const reduxAsyncConnectComponentsLoaded = componentKeys.reduce((loadedMemo, key) => (
```

Comment:
I want a defined logic before any hacking attempt on this. The only reason it doesn't get implemented is that the logic is constantly flawed and doesn't cover some of the client-side cases

@restrry if you have time - would you care to write a proposal for handling filtered async props?
It needs to outline server-side scenario -> just initial load & client-side scenario -> initial load (have partial props from server) & any route change (ie `componentWillReceiveProps` invocation)